### DISCUSSION TITLE
What does DOTNET_PRINT_TELEMETRY_MESSAGE do?

### DISCUSSION THREAD

**Discussion by Bykiev** [dotnet/runtime#95744] [General] [answered]
Hi, in almost all services I see `DOTNET_PRINT_TELEMETRY_MESSAGE` env variable is set to false, but what does it do?

Typical service config:

```
[Unit]
Description=Example .NET Web API App running on Linux

[Service]
WorkingDirectory=/var/www/helloapp
ExecStart=/usr/bin/dotnet /var/www/helloapp/helloapp.dll
Restart=always
# Restart service after 10 seconds if the dotnet service crashes:
RestartSec=10
KillSignal=SIGINT
SyslogIdentifier=dotnet-example
User=www-data
Environment=ASPNETCORE_ENVIRONMENT=Production
Environment=DOTNET_PRINT_TELEMETRY_MESSAGE=false

[Install]
WantedBy=multi-user.target
```

**Comment by gfoidl** [comment]
When you run .NET (SDK) command for the first time, then there's a comment about telemetry printed.
For services this is just noise (and for systemd services this would go to journal), so it's turned off by that env-variable.

**Comment by am11** [comment, marked as answer]
Just like `DOTNET_SKIP_FIRST_TIME_EXPERIENCE`, `DOTNET_PRINT_TELEMETRY_MESSAGE` is basically deprecated in modern .NET.

The correct way to skip the FTUE message is to set `DOTNET_NOLOGO` environment variable to 1.

**Reply by Bykiev** [reply]
Which version of .NET used `DOTNET_PRINT_TELEMETRY_MESSAGE`?

**Reply by am11** [reply]
It is unsupported at least since 2018 https://github.com/dotnet/sdk/issues/9734 (so that would be <= .NET Core 2.1).

**Reply by Bykiev** [reply]
so, do I need to create an issue to update the docs [here]( https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/linux-nginx?view=aspnetcore-8.0&tabs=linux-ubuntu#create-the-service-file)?

**Reply by am11** [reply]
Yes please open an issue in https://github.com/dotnet/AspNetCore.Docs repo. (I recently opened https://github.com/dotnet/docs/issues/37830 for DOTNET_SKIP_FIRST_TIME_EXPERIENCE :D)

**Reply by Bykiev** [reply]
I've created an issue https://github.com/dotnet/AspNetCore.Docs/issues/31221
